search/src/main/scala/weco/api/search/services/FiltersAndAggregationsBuilder.scala

```diff
@@ -34,34 +36,61 @@ trait FiltersAndAggregationsBuilder[Filter, AggregationRequest] {
   val filters: List[Filter]
   val requestToAggregation: AggregationRequest => Aggregation
   val filterToQuery: Filter => Query
-  val searchQuery: Query
 
   def pairedAggregationRequests(filter: Filter): List[AggregationRequest]
+  private def toSelfAggregation(
+    agg: AbstractAggregation,
+    filterQuery: Query
+  ): AbstractAggregation =
+    agg match {
+      case terms: TermsAggregation =>
+        val filterTerm = filterQuery match {
+          case TermsQuery(_, values, _, _, _, _, _) =>
+            values.map(value => s"""\\"$value\\"""").mkString("|")
+          case _ => ""
+        }
+        terms.minDocCount(0).includeRegex(s".*($filterTerm).*")
+      case agg => agg
+    }
+
+  def requestToOrderedAggregation(
```

Comment:
This was another issue I was seeing at the same time - entries in the languages list would jump around when filters are applied.

Because it also makes the main problem a bit easier to test, I have fixed it at the same time.